File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * int val;
 * TreeNode left;
 * TreeNode right;
 * TreeNode() {}
 * TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) {
 * this.val = val;
 * this.left = left;
 * this.right = right;
 * }
 * }
 */
class Solution {
    public int maxDepth(TreeNode root) {

        return findMaxDepth(root, 0);
    }

    private int findMaxDepth(TreeNode node, int depth) {
        if (node == null) return depth;
        return Math.max(findMaxDepth(node.left, depth + 1), findMaxDepth(node.right, depth + 1));

    }
}

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxDepth = function (root) {

    const findDepth = function (node, depth) {
        if (!node) return depth;

        return Math.max(findDepth(node.left, depth + 1), findDepth(node.right, depth + 1));
    }

    return findDepth(root, 0);
};